src/containers/Bookings.js

```diff
@@ -1,70 +1,108 @@
 import React, { Component } from "react";
 import Search from "../components/Search.js";
 import SearchResult from "../components/SearchResult.js";
-import FakeBookings from "../data/fakeBookings.json"
-// import Results from "../components/Results.js";
-// import * as FakeBookings from "../data/fakeBookings.json";
+// import FakeBookings from "../data/fakeBookings.json";
 
 export default class Bookings extends Component {
-  constructor(props){
-    super(props)
+  constructor(props) {
+    super(props);
     this.state = {
-      data : [],
-      show : false,
+      data: [],
+      newData: [],
+      show: false,
+      isLoading: true,
+      err: null,
+      bd: "blue",
       direction: {
-        surname: 'asc',
-        email: 'asc'
+        surname: "asc",
+        email: "asc",
+        firstname: "asc",
+        title: "asc"
       }
-    }
-
+    };
   }
-  
 
-  sortBy = (key) => {
-    console.log(key)
+  toggleHandler = () => {
+    let doesShow = this.state.show;
+    this.setState({
+      data: this.state.newData.rows,
+      show: !doesShow
+    });
+  };
+
+  colourToggleHandler = e => {
+    if (e.currentTarget.className === "Highlite") {
+      e.currentTarget.className = this.state.bd;
```

Comment:
Hmm, this is manipulating the DOM directly. We try to avoid doing this in React because if we re-render after we've manipulated, React will not realise it's changed and remove our changes. We should use state to keep track of whether a row is selected or not.

Reply:
yaaaaaaa Alright! I will use state.